### PR TITLE
fix: check every module when looking if a file is inside the project

### LINKED ISSUE #163: AbstractHandler::isFileInsidePRoject not working when using external modules
If the project structure contains modules that are not contained in the main project folder the sources are considered to be out of the project. 

For instance: 
copilot-pr
  -- copilot-pr
    ----- .idea/
  -- copilot-internal
    ----- tests/flow/

Screenshot 2024-11-18 at 12 10 26

```terminal
2024-11-18 11:58:44,186 [1673639]   WARN - #com.vaadin.plugin.copilot.CopilotPluginUtil - File /Users/vaadin/Workspaces/copilot-pr/copilot-internal/tests/flow/src/main/java/com/vaadin/copilottest/flow/views/helloworld/HelloWorldView.java is not a part of a project
```

### PR DESCRIPTION
**Pull request by mgarciavaadin**
## Description

To check if a file was inside the project we were only checking the project base folder which is not accurate when the project has external modules.
The fix looks in every module in the project including the base project. 

Fixes #163 

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt [hunk 1]
```diff
@@ -153,5 +155,25 @@ class CopilotPluginUtil {
         fun getDotFile(project: Project): VirtualFile? {
             return getDotFileDirectory(project)?.findFile(DOTFILE)
         }
+
+        /**
+         * Returns a map of all base directories related to the project. This includes any external module and the main
+         * project base folders. For the main project, the base directory is the project root and the module name is
+         * "base-module".
+         */
+        fun getBaseDirectoriesForProject(project: Project): Map<String, List<String>> {
```
Comment (MarcinVaadin) [inline src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:164]: Uses same code as in GetModulePathsHandler, might be somehow unified.

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt [hunk 2]
```diff
@@ -45,7 +46,9 @@ abstract class AbstractHandler(val project: Project) : Handler {
     fun isFileInsideProject(project: Project, file: File): Boolean {
         if (file.exists()) {
             val path = file.toPath()
-            return path.toRealPath().startsWith(project.basePath!!)
```
Comment (MarcinVaadin) [inline src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt:48]: Can you try if changing to `project.guessProjectDir()` changes anything?

Comment (mgarciavaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt:48]: You mean using project.guessProjectDir() in my new implementation or trying to fix the whole issue just using the project.guessProjectDir? 
If the second, We would have same problem for external modules located wherever outside the projectDir right?

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt (hunk 1 above)
Comment (mgarciavaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:217]: Moved to CopilotPluginUtil and also data structures to keep them together

File: src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt [hunk 3]
```diff
@@ -153,5 +172,56 @@ class CopilotPluginUtil {
...
+
+            val moduleBaseDirectories = mutableMapOf<String, List<String>>()
+            getModulesInfo(project).forEach { module ->
+                moduleBaseDirectories[module.name] = module.contentRoots.toList()
+            }
+            moduleBaseDirectories["base-module"] = listOf(project.basePath) as List<String>
```
Comment (MarcinVaadin) [inline src/main/kotlin/com/vaadin/plugin/copilot/CopilotPluginUtil.kt:223]: Can we use guessProjectDir here instead of .basePath?

**Review by MarcinVaadin — APPROVED** [review]
Looks ok

**Pull request merged**